src/core/pyproxy.ts

```diff
@@ -2053,7 +2072,12 @@ const PyProxyHandlers = {
     // 2. the result of Python getattr
 
     // python_getattr will crash if given a Symbol.
-    if (jskey in jsobj || typeof jskey === "symbol") {
+    if (
+      typeof jskey === "symbol" ||
+      (jskey in jsobj &&
+        (!(jsobj instanceof Function) ||
+          (jskey !== "name" && jskey !== "length")))
```

Comment:
repeated: I suggest this check is factored out as utility such as `isKeyAcceptable` or something